### PR TITLE
squash: Fix compile errors on pi

### PR DESCRIPTION
**Pull request by popcornmix**

### CODE REVIEW

File: xbmc/cores/VideoPlayer/VideoRenderers/RenderManager.cpp [hunk 1]
```diff
@@ -669,12 +670,6 @@ void CRenderManager::CreateRenderer()
       m_pRenderer = new CRendererMediaCodec;
 #endif
     }
-    if (m_format == RENDER_FMT_MMAL)
```
Comment (FernetMenta) [inline xbmc/cores/VideoPlayer/VideoRenderers/RenderManager.cpp]: what is the purpose of RENDER_FMT_MMAL? can we drop it?
if MMAL is similar to DXVA that serves hw and sw decoding, i would leave it here as well.

Comment (popcornmix) [inline xbmc/cores/VideoPlayer/VideoRenderers/RenderManager.cpp]: MMALRender supports RENDER_FMT_MMAL (opaque handle decoded by MMALCodec), RENDER_FMT_YUV (software decode by ffmeg) and RENDER_FMT_BYPASS (omxplayer where renderer doesn't see individual frames).

Comment (FernetMenta) [inline xbmc/cores/VideoPlayer/VideoRenderers/RenderManager.cpp]: ok, so similar to dxva right? I would keep those lines to indicate that this format is a hw decoder/renderer pair.

**Comment by popcornmix** [comment]
Updated. You mean with MMALRenderer created in two places?

**Comment by popcornmix** [comment]
Note: with this PR omxplayer and software decode are working again, but MMAL has been broken.
https://github.com/FernetMenta/xbmc/blob/master/xbmc/cores/VideoPlayer/VideoPlayerVideo.cpp#L181
is being called before MMAL render is initialised so the opaque pointer of CRenderInfo is NULL.
Any suggestion for how best to fix that?

**Comment by FernetMenta** [comment]
create renderer on RENDER_FMT_NONE if HAS_MMAL

**Comment by FernetMenta** [comment]
fixes also sw decode on Linux.

**Pull request merged**